crates/pathfinder/src/state/sync.rs

```diff
@@ -81,13 +81,15 @@ where
     })?;
 
     // Start update sync-status process.
-    let starting_block = l2_head
-        .map(|(_, hash)| hash)
-        .unwrap_or(StarknetBlockHash(StarkHash::ZERO));
+    let (starting_block_num, starting_block_hash) = l2_head.unwrap_or((
+        StarknetBlockNumber::GENESIS,
```

Comment:
In chat proposed u64::MAX (all fs) here to separate it from first block, which would be more in line with the StarkHash::ZERO which isn't a real block either.